src/transformers/GIPHY.js

```diff
@@ -0,0 +1,39 @@
+import { URL } from 'url';
+
+import { getTrimmedPathName, fetchOEmbedData } from './utils';
+
+const isMediaSubDomain = host => /^(media([0-9]{1,})?\.)giphy\.com$/.test(host);
+
+export const shouldTransform = url => {
+  const { host, pathname } = new URL(url);
+  const trimmedPathName = getTrimmedPathName(pathname);
+
+  return (
+    (['www.giphy.com', 'giphy.com'].includes(host) &&
+      trimmedPathName.includes('gifs/')) ||
+    (isMediaSubDomain(host) && trimmedPathName.includes('media/'))
+  );
+};
+
+export const getGIPHYId = url => {
+  const { host, pathname } = new URL(url);
+
+  if (isMediaSubDomain(host)) {
+    return pathname.split('/')[2];
+  }
+
+  return pathname.split('-').pop();
+};
+
+export const getGIPHYResponsivePadding = ({ height, width }) =>
+  Math.floor((height / width) * 100);
```

Comment:
This should be `round` instead

```suggestion
  Math.round((height / width) * 100);
```